portal/src/graphql/adminapi/UserDetailsConnectedIdentities.tsx

```diff
@@ -273,44 +398,86 @@ const UserDetailsConnectedIdentities: React.FC<UserDetailsConnectedIdentitiesPro
       }
       return (
         <IdentityListCell
+          identityID={item.id}
           identityType="username"
           identityName={item.username}
           addedOn={item.addedOn}
+          onRemoveClicked={onRemoveClicked}
         />
       );
     },
-    []
+    [onRemoveClicked]
   );
 
-  const addIdentitiesMenuProps: IContextualMenuProps = useMemo(
-    () => ({
-      items: [
-        {
-          key: "email",
-          text: renderToString("UserDetails.connected-identities.email"),
-          iconProps: { iconName: "Mail" },
-          onClick: () => navigate("./add-email"),
-        },
-        {
-          key: "phone",
-          text: renderToString("UserDetails.connected-identities.phone"),
-          iconProps: { iconName: "CellPhone" },
-          onClick: () => navigate("./add-phone"),
-        },
-        {
-          key: "username",
-          text: renderToString("UserDetails.connected-identities.username"),
-          iconProps: { iconName: "Accounts" },
-          onClick: () => navigate("./add-username"),
-        },
-      ],
+  const addIdentitiesMenuProps: IContextualMenuProps = useMemo(() => {
+    const availableMenuItem = [
+      {
+        key: "email",
+        text: renderToString("UserDetails.connected-identities.email"),
+        iconProps: { iconName: "Mail" },
+        onClick: () => navigate("./add-email"),
+      },
+      {
+        key: "phone",
+        text: renderToString("UserDetails.connected-identities.phone"),
+        iconProps: { iconName: "CellPhone" },
+        onClick: () => navigate("./add-phone"),
+      },
+      {
+        key: "username",
+        text: renderToString("UserDetails.connected-identities.username"),
+        iconProps: { iconName: "Accounts" },
+        onClick: () => navigate("./add-username"),
+      },
+    ];
+    const enabledItems = availableMenuItem.filter((item) => {
+      return availableLoginIdIdentities.includes(item.key);
+    });
+    return {
+      items: enabledItems,
       directionalHintFixed: true,
-    }),
-    [renderToString, navigate]
-  );
+    };
+  }, [renderToString, navigate, availableLoginIdIdentities]);
 
   return (
     <div className={styles.root}>
+      {confirmationDialogData && (
```

Comment:
It seems the dismiss animation is not there. I suppose we should use `hidden` prop to close the dialog properly.

Reply:
Didn't realise that there is animation, will update this